lib/dry/web/roda/generate.rb

```diff
@@ -0,0 +1,50 @@
+require "pathname"
+require "thor"
+
+module Dry
+  module Web
+    module Roda
+      class Generate
+        SKELETONS_DIR = "skeletons".freeze
+
+        class Processor < Thor
+          include Thor::Actions
+        end
+
+        attr_reader :source_dir
+        attr_reader :processor
+
+        def initialize(skeleton_name)
+          @source_dir = Pathname(__dir__).join(SKELETONS_DIR).join(skeleton_name)
+
+          # TODO: determine if this is a problem across multiple instances
+          @processor = Processor.new
+          @processor.class.source_root source_dir
+        end
+
+        def call(target_dir, scope = {})
+          target_dir = Pathname.getwd + target_dir
+          source_files = Dir[source_dir.join("**/{.,}*")]
+
+          source_files.select { |f| File.file?(f) }.each do |source_file|
```

Comment:
@AMHOL yeah, this is a good idea. I'd like to make something like this accessible to skeleton/generator authors so they don't have to repeat this sort of basic work.

However, I'm not sure I'll get that part done before RDRC, and I'd like to ship at least a basic project generator for dry-web+roda. You OK if I start with this and investigate more general options afterwards? :)